scripts/update-crates-versions.mjs

```diff
@@ -0,0 +1,283 @@
+#!/usr/bin/env node
+
+/**
+ * Script to update versions of publishable Biome crates
+ * Usage: node scripts/update-crate-versions.js <new-version> [--dry-run]
+ * Example: node scripts/update-crate-versions.js 0.6.0
+ * Example (dry-run): node scripts/update-crate-versions.js 0.6.0 --dry-run
+ */
+
+import fs from "node:fs";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+import { parseArgs } from "node:util";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+// Matches a version field in Cargo.toml, capturing the field name with whitespace
+// and the version value. Supports any whitespace/tabs around the equals sign.
+// Example matches: version = "0.5.7", version      = "1.0.0"
+// Capture groups: $1 = 'version\s*=\s*', $2 = version value
+const CARGO_VERSION_FIELD_REGEX = /^(version\s*=\s*)"([^"]+)"$/gm;
+
+// Matches a workspace dependency line in root Cargo.toml for a specific crate.
+// This is dynamically constructed for each crate name.
+// Example: biome_analyze = { version = "0.5.7", path = "./crates/biome_analyze" }
+// Template: `^(${crateName}\\s*=\\s*\\{\\s*version\\s*=\\s*)"([^"]+)"`
+// Capture groups: $1 = everything before version value, $2 = version value
+const WORKSPACE_DEPENDENCY_REGEX_TEMPLATE = (crateName) =>
+	`^(${crateName}\\s*=\\s*\\{\\s*version\\s*=\\s*)"([^"]+)"`;
+
+// Validates simple semantic version format (X.Y.Z only, no pre-release tags)
+// Examples: 0.6.0, 1.0.0, 2.1.3
+// Does NOT accept: 2.0.0-rc.2, 1.0.0-beta.1
+const SEMVER_REGEX = /^\d+\.\d+\.\d+$/;
+
+// List of publishable crates (those that have a version in workspace dependencies)
+const PUBLISHABLE_CRATES = [
+	"biome_analyze",
+	"biome_aria",
+	"biome_aria_metadata",
+	"biome_console",
+	"biome_control_flow",
+	"biome_css_analyze",
+	"biome_css_factory",
+	"biome_css_formatter",
+	"biome_css_parser",
+	"biome_css_syntax",
+	"biome_deserialize",
+	"biome_deserialize_macros",
+	"biome_diagnostics",
+	"biome_diagnostics_categories",
+	"biome_diagnostics_macros",
+	"biome_formatter",
+	"biome_fs",
+	"biome_graphql_analyze",
+	"biome_graphql_factory",
+	"biome_graphql_formatter",
+	"biome_graphql_parser",
+	"biome_graphql_syntax",
+	"biome_grit_factory",
+	"biome_grit_parser",
+	"biome_grit_syntax",
+	"biome_html_factory",
+	"biome_html_parser",
+	"biome_html_syntax",
+	"biome_js_analyze",
+	"biome_js_factory",
+	"biome_js_formatter",
+	"biome_js_parser",
+	"biome_js_semantic",
+	"biome_js_syntax",
+	"biome_js_type_info",
+	"biome_js_type_info_macros",
+	"biome_jsdoc_comment",
+	"biome_json_analyze",
+	"biome_json_factory",
+	"biome_json_formatter",
+	"biome_json_parser",
+	"biome_json_syntax",
+	"biome_markup",
+	"biome_package",
+	"biome_parser",
+	"biome_project_layout",
+	"biome_rowan",
+	"biome_rule_options",
+	"biome_string_case",
+	"biome_suppression",
+	"biome_text_edit",
+	"biome_text_size",
+	"biome_unicode_table",
+];
```

Comment:
It would be nice if we can use read the `publish` property in `Cargo.toml` for each crate as the single source of truth.

Reply:
Yeah, I wanted to reduce FS operations and do crawling, but it should be easy to change it and use "fixed" paths.